api/v1_metrics_apps.go

```diff
@@ -28,20 +30,49 @@ func (app *ApiServer) v1MetricsApps(c *fiber.Ctx) error {
 		return err
 	}
 
-	metrics, err := app.queries.GetAggregateAppMetrics(c.Context(), dbv1.GetAggregateAppMetricsParams{
-		TimeRange: routeParams.TimeRange,
-		LimitVal:  int32(queryParams.Limit),
+	var dateRangeClause string
+	switch routeParams.TimeRange {
+	case "week":
+		dateRangeClause = "date >= CURRENT_DATE - INTERVAL '7 days' AND date < CURRENT_DATE"
+	case "month":
+		dateRangeClause = "date >= CURRENT_DATE - INTERVAL '30 days' AND date < CURRENT_DATE"
+	case "year":
+		dateRangeClause = "date >= CURRENT_DATE - INTERVAL '365 days' AND date < CURRENT_DATE"
+	default: // all_time
+		dateRangeClause = "date < CURRENT_DATE"
+	}
+
+	sql := fmt.Sprintf(`
+		SELECT 
+			COALESCE(developer_apps.name, app_name) AS name,
+			SUM(request_count) AS count
+		FROM api_metrics_apps
+		LEFT JOIN developer_apps ON developer_apps.address = api_metrics_apps.api_key
+		WHERE %s
+		GROUP BY COALESCE(developer_apps.name, app_name)
+		ORDER BY count DESC
+		LIMIT @limit
+	`, dateRangeClause)
+
+	rows, err := app.pool.Query(c.Context(), sql, pgx.NamedArgs{
+		"limit": queryParams.Limit,
 	})
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to query app metrics: %w", err)
 	}
+	defer rows.Close()
 
-	result := make([]AppMetric, len(metrics))
-	for i, metric := range metrics {
-		result[i] = AppMetric{
-			Name:  metric.Name,
-			Count: metric.Count,
+	result := []AppMetric{}
+	for rows.Next() {
```

Comment:
Any reason you're preferring the manual iteration/scan here or is it just what the AI chose?